pytorch_lightning/plugins/training_type/single_device.py

```diff
@@ -8,6 +8,10 @@ def __init__(self, device):
         super().__init__()
         self.device: torch.device = device
 
+    @property
+    def on_tpu(self):
+        return self.device.type == 'xla'
```

Comment:
We should probably check if the xla device is a TPU as `xla` can also mean `GPU`.
https://pytorch.org/xla/release/1.7/index.html#torch_xla.core.xla_model.xla_device_hw

Reply:
I'll remove this